js/ui/components/clubhouse.js

```diff
@@ -55,6 +55,82 @@ const ClubhouseIface =
   </interface> \
 </node>';
 
+var ClubhouseWindowTracker = new Lang.Class({
+    Name: 'ClubhouseWindowTracker',
+
+    _init: function() {
+        this.parent();
+
+        this._windowActor = null;
+
+        // If the Clubhouse is open and the Shell is restarted, then global.get_window_actors()
+        // will still not have any contents when components are initialized, and there's no "map"
+        // signal emitted from the Window Manager either. So instead of relying on the map signal
+        // (which would allow us to track windows individually instead of always having to check the
+        // current list of windows) we connect to the screen's "restacked" signal.
+        this._restackedHandler = global.screen.connect('restacked',
+                                                       this._trackWindowActor.bind(this));
+
+        this._overviewShowingHandler =
+            Main.overview.connect('showing', () => { this.emit('window-changed'); });
+        this._overviewHiddenHandler =
+            Main.overview.connect('hidden', () => { this.emit('window-changed'); });
+    },
+
+    _getClubhouseActorFromWM: function() {
+        return global.get_window_actors().find((actor) => {
+            return (actor.meta_window.get_gtk_application_id() == CLUBHOUSE_ID);
+        });
+    },
+
+    _trackWindowActor: function() {
+        let actor = this._getClubhouseActorFromWM();
+        if (!actor)
+            return;
+
+        if (this._windowActor == actor)
+            return;
+
+        // Reset the current window actor
+        this._untrackWindowActor();
+
+        this._windowActor = actor;
+
+        this.emit('window-changed');
+
+        // Track Clubhouse's window actor to make the closeButton always show on top of
+        // the Clubhouse's window edge
+        this._notifyHandler = actor.connect('notify::x', () => {
+            this.emit('window-changed');
+        });
+
+        this._destroyHandler = actor.connect('destroy', () => {
+            this._untrackWindowActor();
+            this.emit('window-changed');
+        });
+    },
+
+    _untrackWindowActor: function() {
+        if (!this._windowActor)
+            return;
+
+        // Reset the current window actor
+        this._windowActor.disconnect(this._notifyHandler);
+        this._notifyHandler = 0;
+
+        this._windowActor.disconnect(this._destroyHandler);
+        this._destroyHandler = 0;
+
+        this._windowActor = null;
+    },
+
+    getWindowX: function() {
+        if (Main.overview.visible || this._windowActor == null)
+            return -1;
+        return this._windowActor.x;
+    },
+});
+Signals.addSignalMethods(ClubhouseWindowTracker.prototype);
```

Comment:
Inherit from `GObject.Object`, add signals [like this](https://gitlab.gnome.org/GNOME/gnome-shell/blob/master/js/ui/layout.js#L181), and remove this `Signals.addSignalMethods(...);` line.

Reply:
I had that but then I decided to use the `addSignalMethods` instead because I think it's more compact. Is there a reason why using this is less recommendable (so I understand what's better for next time)?